### PR TITLE
README 📝 add details for custom DNS provider(s) + renamed deprecated environment variable 'DNSx' to 'PIHOLE_DNS_x'

### PR DESCRIPTION
**Pull request by thomasmerz**
This PR add some more details for custom DNS provider(s) and adds some help how to choose the fastest and most reliable one (after you've chosen some that care for "independence and privacy"). One deprecated environment variable (`DNSx`) is also renamed to the new variable (`PIHOLE_DNS_x`).

### CODE REVIEW

File: docker-compose.yml [hunk 1]
```diff
@@ -65,8 +65,8 @@ services:
       TZ: "America/Los_Angeles"
       WEBPASSWORD: "" # Blank password - Can be whatever you want.
       ServerIP: 10.2.0.100 # Internal IP of pihole
-      DNS1: 10.2.0.200 # Unbound IP
-      DNS2: 10.2.0.200 # If we don't specify two, it will auto pick google.
+      PIHOLE_DNS_1: 10.2.0.200 # Unbound IP
```
Comment (IAmStoxe) [inline docker-compose.yml:68]: This should not over-write DNS as this is used by the container, not pihole.

File: docker-compose.yml [hunk 2]
```diff
@@ -65,8 +65,8 @@ services:
...
+      PIHOLE_DNS_2: 10.2.0.200 # If we don't specify two, it will auto pick google.
```
Comment (IAmStoxe) [inline docker-compose.yml:69]: Same here

**Review by IAmStoxe — CHANGES_REQUESTED** [review]

File: docker-compose.yml (hunk 1 above)
Comment (thomasmerz) [reply docker-compose.yml:68]: This is exactly my setup with https://github.com/pi-hole/docker-pi-hole and `docker-compose` and it works as expected:
```
etc-pihole/setupVars.conf
13:PIHOLE_DNS_1=45.90.28.39
14:PIHOLE_DNS_2=45.90.30.39
15:PIHOLE_DNS_3=2a07:a8c0::75:86b2
16:PIHOLE_DNS_4=2a07:a8c1::75:86b2
```

Please have a look at:
https://github.com/pi-hole/docker-pi-hole#optional-variables
https://github.com/pi-hole/docker-pi-hole#deprecated-environment-variables

File: docker-compose.yml (hunk 2 above)
Comment (thomasmerz) [reply docker-compose.yml:69]: see above.

**Pull request closed without merging**